pkg/controller/modeladapter/modeladapter_controller.go

```diff
@@ -146,6 +146,12 @@ func newReconciler(mgr manager.Manager, runtimeConfig config.RuntimeConfig) (rec
 		return nil, err
 	}
 
+	klog.Info("Waiting for caches to sync")
+	if ok := cacher.WaitForCacheSync(context.TODO()); !ok {
+		return nil, fmt.Errorf("failed to sync cache")
```

Comment:
if you do not have any variables. I suggest to use `errors.New(..)` instead of `fmt.Errorf(..)`